src/apis/nimbus.js

```diff
@@ -104,6 +120,93 @@ var nimbus = class extends ExtensionAPI {
           },
+
+          async evaluateJEXL(expression, context = {}) {
+            try {
+              const targetingContext = await new TargetingContext(context, {});
+
+              const result = await targetingContext.evalWithDefault(expression);
+
+              return result;
+            } catch (error) {
+              console.error("Error evaluating expression:", error);
+              throw error;
+            }
+          },
+
+          async getClientContext() {
+            function isDate(object) {
+              try {
+                Date.prototype.getDate.call(object);
+                return true;
+              } catch (error) {
+                return false;
+              }
+            }
+
+            async function resolve(object) {
+              if (typeof object === "object" && object !== null) {
+                if (Array.isArray(object)) {
+                  return Promise.all(
+                    object.map(async (item) => resolve(await item)),
+                  );
+                }
+
+                if (isDate(object)) {
+                  return object;
+                }
+
+                const target = {};
+                const promises = Object.entries(object).map(
+                  async ([key, value]) => {
+                    try {
+                      let resolvedValue = await resolve(await value);
+                      return [key, resolvedValue];
+                    } catch (error) {
+                      console.warn(`Error resolving ${key}: `, error);
+                      throw error;
+                    }
+                  },
+                );
+                for (const { status, value } of await Promise.allSettled(
+                  promises,
+                )) {
+                  if (status === "fulfilled") {
+                    const [key, resolvedValue] = value;
+                    target[key] = resolvedValue;
+                  }
+                }
+                return target;
+              }
+
+              return object;
+            }
+
+            const environment = await resolve(
+              lazy.ASRouterTargeting.Environment,
+            );
+            const localContext = await resolve(
+              ExperimentManager.createTargetingContext(),
+            );
+
+            const targetingParameters = {
+              ...environment,
+              ...localContext,
+              locale: lazy.ASRouterTargeting.Environment.locale,
+              localeLanguageCode:
+                lazy.ASRouterTargeting.Environment.localeLanguageCode,
+              region: lazy.ASRouterTargeting.Environment.region,
+              userId: lazy.ClientEnvironment.userId,
+              version: AppConstants.MOZ_APP_VERSION_DISPLAY,
+              channel:
+                lazy.TelemetryEnvironment.currentEnvironment.settings.update
+                  .channel,
+              platform: AppConstants.platform,
+              os: lazy.ClientEnvironmentBase.os,
```

Comment:
We should add a comment that this comes from https://searchfox.org/mozilla-central/source/toolkit/components/messaging-system/targeting/Targeting.sys.mjs#31, but we cant use it because its not exported.


I'll file a bug to export that and then we can conditionally import it in a future version or fall back to specifying all these manually (because the list may change in future versions).